packages/integration-karma/test/mixed-shadow-mode/synthetic-behavior/index.spec.js

```diff
@@ -0,0 +1,90 @@
+import { createElement, setFeatureFlagForTest } from 'lwc';
+import { extractDataIds, isNativeShadowRootInstance } from 'test-utils';
+import ParentAnyChildAny from 'x/parentAnyChildAny';
+import ParentAnyChildReset from 'x/parentAnyChildReset';
+import ParentResetChildAny from 'x/parentResetChildAny';
+import ParentResetChildReset from 'x/parentResetChildReset';
+import ParentLightChildAny from 'x/parentLightChildAny';
+import ParentLightChildReset from 'x/parentLightChildReset';
+
+// In compat mode, native components and synthetic components will both render
+// in synthetic style; there's no difference.
+if (!process.env.NATIVE_SHADOW && !process.env.COMPAT) {
+    describe('synthetic behavior', () => {
+        beforeEach(() => {
+            setFeatureFlagForTest('ENABLE_MIXED_SHADOW_MODE', true);
+        });
+        afterEach(() => {
+            setFeatureFlagForTest('ENABLE_MIXED_SHADOW_MODE', false);
+        });
+
+        const scenarios = [
+            {
+                Component: ParentAnyChildAny,
+                tagName: 'x-parent-any-child-any',
+                nativeLeaf: true,
+            },
+            {
+                Component: ParentAnyChildReset,
+                tagName: 'x-parent-any-child-reset',
+                nativeLeaf: true,
+            },
+            {
+                Component: ParentResetChildAny,
+                tagName: 'x-parent-reset-child-any',
+                nativeLeaf: true,
+            },
+            {
+                Component: ParentResetChildReset,
+                tagName: 'x-parent-reset-child-reset',
+                nativeLeaf: false,
+            },
+            {
+                Component: ParentLightChildAny,
+                tagName: 'x-parent-light-child-any',
+                nativeLeaf: true,
+            },
+            {
+                Component: ParentLightChildReset,
+                tagName: 'x-parent-light-child-reset',
+                nativeLeaf: false,
+            },
+        ];
+
+        scenarios.forEach(({ Component, nativeLeaf, tagName }) => {
+            describe(tagName, () => {
+                let elm;
+                let ids;
+
+                beforeEach(() => {
+                    elm = createElement(tagName, { is: Component });
+                    document.body.appendChild(elm);
+                    ids = extractDataIds(elm);
+                });
+
+                it('should render the shadow root in the correct shadow mode', () => {
+                    expect(isNativeShadowRootInstance(ids.leaf.shadowRoot)).toEqual(nativeLeaf);
+                });
```

Comment:
The reason for this test is that it's actually possible for us to render the shadow root in one shadow mode, and yet treat it as the other mode for scope attributes and ID-mangling. That was the whole bug we ran into.